pkg/fqdn/matchpattern/matchpattern_test.go

```diff
@@ -25,14 +25,28 @@ var _ = Suite(&MatchPatternTestSuite{})
 // cilium.io. -> cilium[.]io[.]
 // *.cilium.io. -> [-a-zA-Z0-9]+.cilium[.]io[.]
 // *cilium.io. -> "([a-zA-Z0-9]+[.])?cilium[.]io[.]
-func (ts *MatchPatternTestSuite) TestMatchPatternREConversion(c *C) {
+func (ts *MatchPatternTestSuite) TestAnchoredMatchPatternREConversion(c *C) {
 	for source, target := range map[string]string{
 		"cilium.io.":   "^cilium[.]io[.]$",
 		"*.cilium.io.": "^" + allowedDNSCharsREGroup + "*[.]cilium[.]io[.]$",
 		"*":            "(^(" + allowedDNSCharsREGroup + "+[.])+$)|(^[.]$)",
```

Comment:
Should this use non-capturing groups?

```suggestion
		"*":            "^(?:(?:" + allowedDNSCharsREGroup + `+\.)+|\.)$`,
```

Reply:
Yeah, I agree. I think its outside the scope of this PR tho., and imo we should just make that regex match everything, ala. `^.*$` or something, since by design it means _match everything. And if so, we can skip the regex `MatchString` call as well.